### ISSUE TITLE
Questions about parameters and the backbone model

### ISSUE THREAD

**Issue by tom68-ll** [Edillower/CycleNLG#1]
"Hello, author. Could you please provide specific hyperparameter settings? Additionally, could you advise me on which model to choose for initializing data-to-text and text-to-data?

**Comment by Edillower** [comment]
Hi, you can check section 4.3 for the training parameters. In our paper, we use T5-base as the backbone model, but you can also use any other seq2seq models that are available throught Huggingface.

**Comment by tom68-ll** [comment]
Thank you very much for your response. I will make another attempt

**Comment by Edillower** [comment]
Feel free to reopen the issue if you run into any further problems.

**Comment by tom68-ll** [comment]
Sorry for bothering again, may I ask if we need to preprocess the dataset to fit this task, like with WebNLG?

**Comment by Edillower** [comment]
See section 4.1 of the paper for preprocessings we have done. Those processings are some conventional things previous works on WebNLG and T5 have done, but you can definitely preprocess the data in a way you prefer though the results may vary a little bit.

**Comment by ktxlh** [comment]
What "prevous works" are you referring to? It is fine to just point us to a specific paper...

**Comment by Edillower** [comment]
Hi @ktxlh,
For (1) Prefixing task description string is what described in the original T5 paper.
For (2) convert camel-cased or snake-cased subjects, predicates and objects to regular strings; and (3) normalize accented characters, there's a convertion script that has been reused by multiple previous works on WebNLG. Let me try to find the original github repo and get back to you later. 

Thanks,
Zhuoer Wang

**Comment by Edillower** [comment]
@ktxlh 
Not sure which is the first or who is the original author, but 
For (2) we used the camel_case_split function: https://github.com/search?q=camel_case_split+webnlg&type=code
For (3) we used the remove_accents function: https://github.com/lizhe2016/RFBFN/blob/9dfcabeb60e6f9583b5691ac48125322c8e354f2/data_preprocess/preprocess.py#L377
Feel free to let me know if you need any further assitance from my side.

**Comment by Edillower** [comment]
@ktxlh  Following the issue you opened at the Amazon repo - Sorry I wasn't able to release the data/preprocessing script via Amazon, but here are some sample inputs for your reference. 

Sample data file (line sperated):
Generate in English: [S] Aarhus Airport [P] location [O] Tirstrup [S] Tirstrup [P] is part of [O] Central Denmark Region [S] Tirstrup [P] is part of [O] Denmark
Generate in English: [S] Aarhus Airport [P] operating organisation [O] Aarhus Lufthavn A/S [S] Aarhus Airport [P] runway name [O] 10L/28R [S] Aarhus Airport [P] runway length [O] 2702.0
Generate in English: [S] Aarhus Airport [P] runway length [O] 2776.0 [S] Aarhus Airport [P] operating organisation [O] Aarhus Lufthavn A/S [S] Aarhus Airport [P] runway name [O] 10L/28R

Sample text file(line sperated):
Extract Triplets: 1634 The Ram Rebellion was written in the US, where Native Americans are an ethnic group, Washington DC is the capital and Barack Obama is the leader and President.
Extract Triplets: The demonym for people living in the United States is Americans, African Americans are an ethnic group there and is where 1634 The Ram Rebellion was written. The leader of the United States is known as the President and is currently Barack Obama.
Extract Triplets: The United States are inhabited by Americans and the ethnic group of African Americans. 1634 The Ram Rebellion was written in the country where President Barack Obama is the leader.
